main/acle.md

```diff
@@ -7499,6 +7626,1415 @@ This is because the ACLE intrinsic calls do not imply a particular
+  __attribute__((arm_streaming_compatible, arm_preserves_za))
+  uint64_t svcntsh();
+
+  // Return the number of words in a streaming vector.
+  // Equivalent to svcntw() when called in streaming mode.
+  __attribute__((arm_streaming_compatible, arm_preserves_za))
+  uint64_t svcntsw();
+
+  // Return the number of doublewords in a streaming vector.
+  // Equivalent to svcntd() when called in streaming mode.
+  __attribute__((arm_streaming_compatible, arm_preserves_za))
+  uint64_t svcntsd();
+```
+
+`svcntsb()` is equivalent to an RDSVL instruction with an immediate
+operand of 1. There are no intrinsics that map directly to other
+immediate operands, or to the ADDSVL and ADDSPL instructions, but it is
+possible to write these operations using normal C arithmetic. For example:
+
+``` c
+  x0 = svcntsb();           // corresponds to RDSVL x0, #1
+  x0 = svcntsb() * 5;       // can be optimized to RDSVL x0, #5
+  x0 = x1 + svcntsb();      // can be optimized to ADDSVL x0, x1, #1
+  x0 = x1 - svcntsb() * 3;  // can be optimized to ADDSVL x0, x1, #-3
+  x0 = x1 + svcntsd();      // can be optimized to ADDSPL x0, x1, #1
+  x0 = x1 - svcntsw();      // can be optimized to ADDSPL x0, x1, #-2
+```
+
+#### SUMOPA
+
+``` c
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsumopa_za32[_s8](uint64_t tile, svbool_t pn, svbool_t pm,
+                          svint8_t zn, svuint8_t zm);
+
+  // Only if __ARM_FEATURE_SME_I16I64 != 0
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsumopa_za64[_s16](uint64_t tile, svbool_t pn, svbool_t pm,
+                           svint16_t zn, svuint16_t zm);
+```
+
+#### SUMOPS
+
+``` c
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsumops_za32[_s8](uint64_t tile, svbool_t pn, svbool_t pm,
+                          svint8_t zn, svuint8_t zm);
+
+  // Only if __ARM_FEATURE_SME_I16I64 != 0
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsumops_za64[_s16](uint64_t tile, svbool_t pn, svbool_t pm,
+                           svint16_t zn, svuint16_t zm);
+```
+
+#### USMOPA
+
+``` c
+  __attribute__((arm_streaming, arm_shared_za))
+  void svusmopa_za32[_u8](uint64_t tile, svbool_t pn, svbool_t pm,
+                          svuint8_t zn, svint8_t zm);
+
+  // Only if __ARM_FEATURE_SME_I16I64 != 0
+  __attribute__((arm_streaming, arm_shared_za))
+  void svusmopa_za64[_u16](uint64_t tile, svbool_t pn, svbool_t pm,
+                           svuint16_t zn, svint16_t zm);
+```
+
+#### USMOPS
+
+``` c
+  __attribute__((arm_streaming, arm_shared_za))
+  void svusmops_za32[_u8](uint64_t tile, svbool_t pn, svbool_t pm,
+                          svuint8_t zn, svint8_t zm);
+
+  // Only if __ARM_FEATURE_SME_I16I64 != 0
+  __attribute__((arm_streaming, arm_shared_za))
+  void svusmops_za64[_u16](uint64_t tile, svbool_t pn, svbool_t pm,
+                           svuint16_t zn, svint16_t zm);
+```
+
+#### ZERO
+
+``` c
+  __attribute__((arm_streaming, arm_shared_za))
```

Comment:
Should the `svzero_*` intrinsics be marked as `arm_streaming_compatible`? The instruction definitions don't require them to be executed in streaming mode (they're predicated with `CheckSMEAndZAEnabled`, not `CheckStreamingSVEAndZAEnabled`

Reply:
Oops, thanks for catching that.  I've also fixed `svldr` and `svstr`.

Reply:
@allcontributors please add @sdesmalen-arm for review.

Reply:
@all-contributors please add @sdesmalen-arm for review.